### PR TITLE
Forum: Bugfixes of PHP 8 Regressions

### PR DESCRIPTION
**Pull request by playamongthestars**

### CODE REVIEW

File: Modules/Forum/classes/class.ilForumTopicTableGUI.php [hunk 1]
```diff
@@ -290,8 +290,6 @@ protected function fillRow(array $a_set) : void
         $this->tpl->setVariable('CSS_ROW', $css_row);
 
         $this->ctrl->setParameter($this->getParentObject(), 'thr_pk', '');
-        $this->ctrl->setParameter($this->getParentObject(), 'user', '');
```
Comment (mjansenDatabay) [inline Modules/Forum/classes/class.ilForumTopicTableGUI.php:293]: Why did you remove this?

File: Modules/Forum/classes/class.ilForumTopic.php [hunk 2]
```diff
@@ -688,9 +688,6 @@ public function getNestedSetPostChildren(?int $pos_id = null, ?int $num_levels =
             $casted_row['post_id'] = (int) $row['post_id'];
             $casted_row['post_read'] = (int) $row['post_read'];
             $casted_row['children'] = (int) $row['children'];
-            $casted_row['rgt'] = (int) $row['rgt'];
```
Comment (mjansenDatabay) [inline Modules/Forum/classes/class.ilForumTopic.php:691]: Are you sure these are never used?

File: Modules/Forum/classes/class.ilForumSettingsGUI.php [hunk 3]
```diff
@@ -762,30 +762,19 @@ private function initNotificationSettingsForm() : bool
 
             $cb_grp = new ilCheckboxGroupInputGUI($this->lng->txt('notification_settings'), 'notification_events');
 
-            $notify_modified = new ilCheckboxInputGUI($this->lng->txt('notify_modified'), 'notify_modified');
-            $notify_modified->setValue((string) ilForumNotificationEvents::UPDATED);
+            $notify_modified = new ilCheckboxOption($this->lng->txt('notify_modified'), (string) ilForumNotificationEvents::UPDATED);
```
Comment (mjansenDatabay) [inline Modules/Forum/classes/class.ilForumSettingsGUI.php:765]: Dind't affect this the storage of the form?

File: Modules/Forum/classes/class.ilForumTopicTableGUI.php (hunk 1 above)
Comment (playamongthestars) [reply Modules/Forum/classes/class.ilForumTopicTableGUI.php:293]: Because, it adds empty parameters to the url which leads to crashes

File: Modules/Forum/classes/class.ilForumTopic.php (hunk 2 above)
Comment (playamongthestars) [reply Modules/Forum/classes/class.ilForumTopic.php:691]: yes

File: Modules/Forum/classes/class.ilForumSettingsGUI.php (hunk 3 above)
Comment (playamongthestars) [reply Modules/Forum/classes/class.ilForumSettingsGUI.php:765]: In fact, I can test this, after the getLinkTarget-issue is fixed.
